rbmc-cfam-daemon/src/cfam_fields.cpp

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+#include "cfam_fields.hpp"
+
+#include <format>
+#include <ranges>
+#include <utility>
+
+namespace cfam
+{
+
+uint32_t getFieldValue(const RegMap& regs, const FieldMetadata& meta)
+{
+    if (!regs.contains(meta.reg))
+    {
+        throw std::invalid_argument{
+            std::format("getFieldValue: reg {} not in RegMap",
+                        std::to_underlying(meta.reg))};
+    }
+    return getFieldValue(regs.at(meta.reg), meta.start, meta.numBits);
+}
+
+uint32_t getFieldValue(uint32_t regData, size_t start, size_t numBits)
+{
+    if ((start + numBits) > 32)
+    {
+        throw std::invalid_argument{
+            std::format("getFieldValue: Start bit {} + numBits {} too big",
```

Comment:
Suggestion: Good to include expect bit length in the trace.

Reply:
good idea.